nixos/modules/services/continuous-integration/hercules-ci-agent/gc.nix

```diff
@@ -0,0 +1,24 @@
+{ lib, pkgs, config, ... }:
+let
+
+  inherit (lib) types mkIf mkOption;
+  cfg = config.services.hercules-ci-agent;
+
+in
+{
+  options.services.hercules-ci-agent = {
```

Comment:
This is not specific to hercules-ci-agent at all - it only enables some `nix.gc` options. If you you feel like more options are needed there, please add them directly to the `nix.gc` namespace.

Reply:
It is required for correct operation though. That's not to say that it can't be done differently, which is why it's opt-in. That's already a change from the existing module.
Another reason for this to exist is that the agent will implement garbage collection logic to purge paths which are unlikely to be used first.

I'll add a comment.

Reply:
If it's required for correct operation, why not assert some of the `nix.gc` options being set, and exposing the `--max-freed` option in the `nix.gc` namespace (and module) as an option?

You shouldn't cross boundaries of this module, and it's something useful for all NixOS users, not just people configuring the `hercules-ci-agent`.

Reply:
It's potentially useful for all NixOS users, but I'm not convinced that this should be the blessed implementation.
I've removed it.